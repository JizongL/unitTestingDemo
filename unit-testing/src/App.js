import React, { Component } from 'react';
import Event from './Event/event'
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        {this.props.message}
        <div>
        <Event onClick={() => console.log('click') } />
        </div>
      </div>
    );
  }
}
export default App;
