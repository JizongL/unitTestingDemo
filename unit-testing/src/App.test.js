import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer'

describe('my app unit test',()=>{
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  
  it('renders the UI as expected',()=> {
    const tree = renderer
    .create(<App message = 'testing'/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
})
