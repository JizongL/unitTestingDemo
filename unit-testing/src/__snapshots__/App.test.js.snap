// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my app unit test renders the UI as expected 1`] = `
<div
  className="App"
>
  testing
  <div>
    <button
      onClick={[Function]}
    >
      Click me!
    </button>
  </div>
</div>
`;
